nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -92,6 +91,188 @@ def cli(ctx, target):
 
     ctx.obj["site_config"] = site_config
 
+
+@cli.command()
+@click.option("--mongo-uri",required=False, default="mongodb://localhost:37020",)
+@click.option( "--no-update", is_flag=True, default=False, help="Do not update the database")
+@click.pass_context
+def update_metabolomics(ctx, mongo_uri, no_update=False):
+    """
+    Update the NMDC Metabolomics Activity and related DataObject records:
+     - update the has_input and has_output data objects
+        - update the parent OmicsProcessing.has_output record with the new has_input data objects
+        - Re-ID the  Metabolomics Activity record and update with the new has_output data objects
+    """
+    start_time = time.time()
+    logging.info(f"Updating NMDC metabolomics data objects")
+    # Connect to the MongoDB server and check the database name
+    client = pymongo.MongoClient(mongo_uri, directConnection=True)
+    with pymongo.timeout(5):
+        assert ("nmdc" in client.list_database_names()), f"Database nmdc not found"
+    db_client = client["nmdc"]
+    # start a session
+    session = client.start_session()
+
+    # API client for minting new IDs
+    config = ctx.obj["site_config"]
+    api_client = NmdcRuntimeApi(config)
+
+    # Keep track of the updated record identifiers
+    updated_record_identifiers = []
+
+    # Get all Metabolomics OmicsProcessing records
+    omics_processing_records = db_client["omics_processing_set"].find(
+        {"omics_type.has_raw_value": "Metabolomics"}
+    )
+    len_omics_processing_records = len(list(omics_processing_records.clone()))
+    logging.info(f"Updating {len_omics_processing_records} Metabolomics OmicsProcessing records")
+    # non-orphan Metabolomics Activity records have a parent OmicsProcessing record
+    non_orphan_metabolomics_ids = []
+    for omics_processing_record in omics_processing_records:
+        omics_legacy_id = omics_processing_record.get("alternative_identifiers", [])[0]
+        omics_new_id = omics_processing_record["id"]
+        logging.info(f"Updating Metabolomics Omics record: {omics_legacy_id} / {omics_new_id}")
+
+        # Find Metabolomics Activity records that are informed_by the legacy OmicsProcessing ID
+        metabolomics_activity_records = db_client["metabolomics_analysis_activity_set"].find(
+            {"was_informed_by": omics_legacy_id}
+        )
+        len_metabolomics_activity_records = len(list(metabolomics_activity_records.clone()))
+        logging.info(f"Found {len_metabolomics_activity_records} Metabolomics Activity records for {omics_legacy_id}")
+
+
+        for metabolomics_activity_record in metabolomics_activity_records:
```

Comment:
metabolomics analysis activity records also need their has_calibration values updated. If the value is 'false' or 'False' do nothing, if the value has an emsl prefix, find that record in the data object collection, update that record and update the metabolomics analysis activity record.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/116/commits/4bf4eb05c60ea1aad1b12501c8ec09ed1e42aa56